### PR TITLE
Feature/router

### PR DESCRIPTION
**Pull request by lorof**

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -28,8 +28,11 @@ const Title = styled.h1`
   margin: 10px 0 45px 0;
 `
 
+interface IHeader {
+  children?: ReactNode
+}
 
-const Header:React.FC = () => {
+const Header:React.FC = ({children}: IHeader) => {
```
Comment (AwesomeObserver) [inline src/components/Header.tsx]: Rewrite to
```ts
const Header:React.FC<IHeader> = ({children})
```

File: src/views/Page.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+`
+
+interface IPage {
+  form?: React.ComponentClass,
+  content?: any,
+  path?: string,
+  default?: boolean
+}
+
+const Page = ({form: Form, content: Content, ...props}: IPage) => {
```
Comment (AwesomeObserver) [inline src/views/Page.tsx]: Rewrite with `FC` alias type

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  transition: background-color .3s;
+  outline: none;
+  &:hover {
+    background-color: #c792ea;
+  }
+`
+
+
+
+interface IProps {
```
Comment (AwesomeObserver) [inline src/components/Button.tsx]: In component, I think to best use `type` instead of `interface`

File: src/components/Header.tsx [hunk 4]
```diff
@@ -39,6 +42,7 @@ const Header:React.FC = () => {
       <HeaderBottom>
         <Container>
           <Title>grep over package repositories</Title>
+          {children ? children : null}
```
Comment (AwesomeObserver) [inline src/components/Header.tsx]:
```ts
{children && children}
```
🤔

File: src/views/Page.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+`
+
+interface IPage {
+  form?: React.ComponentClass,
+  content?: any,
+  path?: string,
+  default?: boolean
+}
+
+const Page:React.FC<IPage> = ({form: Form, content: Content, ...props}) => {
```
Comment (AwesomeObserver) [inline src/views/Page.tsx]: wtf

**Comment by lloss** [comment]
Your current task called "integrate reach router" , all other changes that do not relate to it should be placed in a different branch and it is not very clear what they are doing in this pull-quest

**Pull request closed without merging**